src/synthesiser/Synthesiser.cpp

```diff
@@ -837,7 +837,7 @@ void Synthesiser::emitCode(std::ostream& out, const Statement& stmt) {
                 if (const auto* signedConstant = as<ram::SignedConstant>(constant)) {
                     value = ramBitCast<RamDomain>(signedConstant->getValue());
                 } else if (const auto* stringConstant = as<ram::StringConstant>(constant)) {
-                    value = ramBitCast<RamDomain>(
+                    value = static_cast<RamDomain>(
```

Comment:
This really should be `ramBitCast` (want bit pattern reinterpretation, not numeric coercion), but `convertSymbol2Idx` should return `RamUnsigned` instead of `size_t`.

Reply:
Hi @OlivierHamel, good catch.

I was being lazy and trying to get the compiler to shut up with a static_cast but your suggestion is much better.

Thanks!

Reply:
I've a complementary system that works the other way around: Given statistics and functional dependencies for the EDB (currently passed in as pragma metadata), it estimates join sizes using an an abstract interpretation. This is plugged into a an optimizer (best-N literals exhaustive, followed by greedy).

This can be less precise than `--profile` IDB info, but is useful when developing the program since it only requires statistics on the EDB. Ping me if this is something you think would be useful for you folks. It'll require some modifications to bring upstream (this is the main reason why I don't have a PR for it).

Reply:
Hi @OlivierHamel,

That's really interesting.

I can see the advantage in not having to execute the program (say during rapid prototyping).

Perhaps after this PR is merged you can try out the auto-scheduler in your own system and report back on the performance numbers (both in terms of the auto-scheduling time and the performance of the produced schedules).

Reply:
I'll see if I have time; I'd have to backport it since we're stuck on an old fork of Souffle.